server/gitrest/packages/gitrest-base/src/utils/isomorphicgitManager.ts

```diff
@@ -88,6 +89,9 @@ export class IsomorphicGitRepositoryManager extends RepositoryManagerBase {
 				},
 				err,
 			);
+			if (this.isEphemeralContainer && err?.code === "NotFoundError") {
```

Comment:
If err?.code is 404, we can throw 404 even if it is not an EphemeralContainer.

Reply:
That is a good question. With 404, we are telling the clients to give up on the document and create a new one. This is totally fine for ephemeral containers but it should ideally not happen in the durable containers. I think it comes down to whether we want the clients to retry or give up for durable containers. I think @hedasilv and @znewton might be in a better suited to make that call.

Reply:
@pradeepvairamani is correct about the impact of that type of change. There needs to be a larger effort to cleanup and improve Gitrest's error codes, and I don't think we should increase the scope of this PR with that work.

There are some parts of R11s (including the driver) that rely on specific error codes to operate correctly. For example, `getOrCreateRepository` from services-client explicitly requires a `400`, instead of the more logical and correct `404`, to create a repo when get repo fails. Another example is `GitManager.getRef`, which returns `null` on 400 or 404, or throws otherwise. We need to be careful when changing that error code so that we don't break what was previously working with a 400.

TL;DR: Changing Gitrest error codes should receive a sizeable amount of attention and validation, and I don't think this is the right time/PR to do that.